scene/main/node.cpp

```diff
@@ -437,18 +439,18 @@ void Node::set_process_mode(ProcessMode p_mode) {
 	bool prev_can_process = can_process();
 	bool prev_enabled = _is_enabled();
 
-	data.process_mode = p_mode;
-
-	if (data.process_mode == PROCESS_MODE_INHERIT) {
+	if (p_mode == PROCESS_MODE_INHERIT) {
 		if (data.parent) {
-			data.process_owner = data.parent->data.owner;
+			data.process_owner = data.parent->data.process_owner;
```

Comment:
This was causing a bug when switching back to inherit for a node with at least depth 2 in the tree, it was always using the direct parent as process owner instead of the correct node.